### PR TITLE
feat: add quantity pattern builder

### PR DESCRIPTION
**Pull request by micgro42**

### CODE REVIEW

File: tests/unit/sparql/QueryObjectBuilder.spec.ts [hunk 1]
```diff
@@ -101,9 +101,216 @@ describe( 'QueryObjectBuilder', () => {
 			omitLabels: true,
 		};
 
-		expect( () => {
-			builder.buildFromQueryRepresentation( quantityQueryRepresentation );
-		} ).toThrow();
+		const expected = {
```
Comment (Ladsgroup) [inline tests/unit/sparql/QueryObjectBuilder.spec.ts]: This is massive, can you move it to a json file or another js script?

File: src/sparql/SyntaxBuilder.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import {
 	BgpPattern,
-	BlankTerm,
-	IriTerm,
+	BlankTerm, FilterPattern,
```
Comment (Ladsgroup) [inline src/sparql/SyntaxBuilder.ts]: nitpick: Can all be either one per line or filled as much as possible not somewhere in between?

File: src/sparql/QuantityValuePatternBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+		return patterns;
+	}
+
+	private isQuantityValue( value: string | UnitValue ): value is UnitValue {
+		return typeof value === 'object' &&
+			( ( value as UnitValue ).unit === null || typeof ( value as UnitValue ).unit === 'string' );
+	}
+
+	private getFilterOperatorFromRelation( propertyValueRelation: PropertyValueRelation ): string | null {
```
Comment (Ladsgroup) [inline src/sparql/QuantityValuePatternBuilder.ts:251]: This probably can be moved to a central place but can happen later.

File: src/sparql/QuantityValuePatternBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+		numericValue: number,
+		unit: string,
+		conditionIndex: number,
+		propertyId: string,
+		referenceRelation: ReferenceRelation,
+		propertyValueRelation: PropertyValueRelation,
+	): Pattern[] {
+		const patterns: Pattern[] = [];
+
+		const userQuantityVariable = SyntaxBuilder.buildVariableTermFromName( 'userQuantity' );
```
Comment (Ladsgroup) [inline src/sparql/QuantityValuePatternBuilder.ts]: Suggestion for refactoring for later. Inject SyntaxBuilder similar to TripleBuilder to make dependencies of this class clear.

File: tests/unit/sparql/QueryObjectBuilder.spec.ts (hunk 1 above)
Comment (micgro42) [reply tests/unit/sparql/QueryObjectBuilder.spec.ts]: And it was a pain to manually write, let me tell you that. I'm still looking into options here.

File: src/sparql/SyntaxBuilder.ts (hunk 2 above)
Comment (micgro42) [reply src/sparql/SyntaxBuilder.ts]: No idea why my IDE didn't do that by itself. Fixed.

File: src/sparql/QuantityValuePatternBuilder.ts (hunk 3 above)
Comment (micgro42) [reply src/sparql/QuantityValuePatternBuilder.ts:251]: Let's wait for `DateValuePatternBuilder` to land to see what options we're really dealing with.

File: src/sparql/QuantityValuePatternBuilder.ts (hunk 4 above)
Comment (micgro42) [reply src/sparql/QuantityValuePatternBuilder.ts]: Done as a consequence of doing it for one of the previous commits in the chain

**Comment by micgro42** [comment]
@Ladsgroup was right for wanting to keep that jest env setup in #213. I just now added it again, after I deleted it in that commit :facepalm:

File: tests/unit/sparql/QueryObjectBuilder.spec.ts (hunk 1 above)
Comment (Ladsgroup) [reply tests/unit/sparql/QueryObjectBuilder.spec.ts]: :( let me know if you found an option.

Comment (micgro42) [reply tests/unit/sparql/QueryObjectBuilder.spec.ts]: I extracted it into a .json file. What do you think?

**Review by Ladsgroup — APPROVED** [review]
Legend

**Comment by micgro42** [comment]
Thanks!

**Pull request merged**